### PR TITLE
Add Dodgems Style

### PR DESCRIPTION
**Pull request by karst**
Add fall back song for the Dodgems beat style.

Feedback is welcome!

### CODE REVIEW

**Comment by mattheasboelter** [comment]
Looks good to me! We should be set to merge.

**Comment by IntelOrca** [comment]
I suggest using FLAC rather than WAV.

**Comment by karst** [comment]
I personally don't really see the point. Both WAV and FLAC are lossless audio. I rather use OGG when it comes to the final release as you don't really need to audiophile level quality in the game (as the game itself processes the audio, which reduces the audio quality anyways). Perhaps make a WAV/FLAC pack for high quality audio for people that want it, but I rather try keeping to keep the file size low if we are gonna change the format from WAV (which is currently the only accepted by the games engine).

**Comment by mattheasboelter** [comment]
I agree with this. If we decide we want everything to be flac, wav gives us enough information to make that conversion. Until we have time to actually make a proper decision about that, I don't really see a reason to worry about it as long as we're receiving lossless files. At the moment, we're really only just barely starting to get progress on music again. I think it's best to really just focus on the creative aspects of making that happen, and then worry about the more technical details once we have more things solidified.

I think we should merge this pull request as is.

**Comment by karst** [comment]
Just talked with IO over Discord DM and they explained that FLAC is slightly smaller and if we want to keep the repository as small as possible with lossless audio we are better off using FLAC. If we want to switch to WAV we can convert FLAC back to that, or if we want to smallen it we can move to OGG Vorbis.

**Comment by ExileFox** [comment]
WAV has no benefit over FLAC other than that it would work on ancient media players like old iPods, Windows Media Player and such without additional codecs.

File: other_styles/dodgem/object.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "id": "auraspecs.music.DODGEM",
+    "authors": [
+        "Karst van Galen Last"
+    ],
+    "version": "1.0",
+    "objectType": "music",
+    "properties": {
+        "tracks": [
+            {
+                "source": "DODGEM.wav",
```
Comment (Gymnasiast) [inline other_styles/dodgem/object.json]: Please rename the file to `dodgems.flac`. Rename the directory as well. And also change this line.

**Comment by karst** [comment]
What do you guys prefer to have in the Author name? My artist name, my in game name or my real name?

**Comment by tupaschoal** [comment]
> What do you guys prefer to have in the Author name? My artist name, my in game name or my real name?

Just my opinion: whichever you feel like using

Not sure what other team members think 😛

**Comment by Gymnasiast** [comment]
Your in-game name is probably not too relevant.

I think it would be a good idea to include your real name, but you can use include both, e.g. `Jalmaan (Karst van Galen Last)`.

File: other_styles/dodgem/object.json (hunk 1 above)
Comment (IntelOrca) [reply other_styles/dodgem/object.json]: @Gymnasiast are you sure? Given we don't have flac support yet, I was going to convert to .wav on the CI when generating the parkobj. On the other hand, having flac extension even though it is still a .wav would probably still work.

Comment (Gymnasiast) [reply other_styles/dodgem/object.json]: > On the other hand, having flac extension even though it is still a .wav would probably still work.

It is not a .wav any more. Karst replaced the wav with a .flac file, but didn't update the json file initially.

Comment (IntelOrca) [reply other_styles/dodgem/object.json]: You misunderstand me. I want it as .flac to keep the repository size down. The CI can then convert it to .wav and create a .parkobj that works in OpenRCT2. We don't yet support .flac, so it has to be a .wav for this .parkobj to work in NSF.

**Comment by karst** [comment]
Anything that still needs to happen?

**Comment by Gymnasiast** [comment]
I discussed some things with IntelOrca and this is how we’re going to tackle it:
- You commit the music as flac to the repository
- The filename in the object should still reference the .wav version
- When distributing the objects, we will convert the flac to .wav (as long as OpenRCT2 requires it)
- Use lower case extensions for all filenames.

**Comment by Gymnasiast** [comment]
Btw, why did you call this style "bumpercar style"? They’re called dodgems in British English. Or is it just to differentiate them from the vanilla versions?

**Comment by karst** [comment]
I have called it bumpercar style to differentiate from the vanilla version indeed. :)

And ill make sure to keep on that format and change my stuff to that format accordingly. You want me to submit it here in just FLAC and you guys convert it to WAV?

And do I need to submit music to the object directory? As I have with Galaxy Style

**Comment by Gymnasiast** [comment]
> I have called it bumpercar style to differentiate from the vanilla version indeed. :)

> And ill make sure to keep on that format and change my stuff to that format accordingly. You want me to submit it here in just FLAC and you guys convert it to WAV?

We may want to change the names still, but that’s easy and cheap (just a few bytes, Git will have no problem with that). Let’s get this one in the repo ;-)

> And do I need to submit music to the object directory? As I have with Galaxy Style

I guess we should use the same structure as we would in the objects directory. Namely:

- replacements/openrct2.music.bumpercar/object.json
- replacements/openrct2.music.bumpercar/music/0.flac

**Comment by karst** [comment]
Okay! I'll do that

**Comment by IntelOrca** [comment]
Yeah, best not to call it Bumpercar style as that may be the US translation for Dodgems beat.

**Comment by karst** [comment]
Yea, all the names I have put in are suggestions. They're not necessary final :)

**Comment by Gymnasiast** [comment]
@karst This should now reflect what I _think_ Ted was asking for. If he approves the PR, I will merge it.

**Review by IntelOrca — APPROVED** [review]

**Pull request merged**